### DISCUSSION TITLE
Change the default action for clicking on table rows to "view"

### DISCUSSION THREAD

**Discussion by holymp2006** [filamentphp/filament#8628] [Help]
I have created a custom view page for table rows and need to change the default click action for each row to navigate to this view page.

Here's what I've tried:

I added `'view' => Pages\ViewOrder::route('/{record}'),` to the `static::getPages()` function and included a `url()` method for each column to point to this route. However, this resulted in link formatting being applied, changing the color of the cell's text to the primary color. I do not want this behavior.

The default edit url does not apply this formatting.

Any ideas, please?

**Comment by mattlibera** [comment]
try using `recordUrl()` on your `$table` declaration:

```
return $table
    // .... your schema...
    ->recordUrl(<your record URL here>)
```

you may also need to use `->recordAction(null)` to unset the default record action.

**Comment by yuvrajtimalsina** [comment]
```
 ->recordUrl(fn (Model $record): string => YourResource::getUrl('view', ['record' => $record]))
 ```
